File: app/src/main/java/com/esafirm/androidplayground/conductor/ConductorSample.java
package com.esafirm.androidplayground.conductor;

import android.graphics.Color;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.bluelinelabs.conductor.Controller;

import java.io.Serializable;

public class ConductorSample extends Controller implements Serializable {

    @NonNull
    @Override
    protected View onCreateView(@NonNull LayoutInflater inflater, @NonNull ViewGroup container) {
        TextView textView = new TextView(container.getContext());
        textView.setText("Hello World");
        textView.setTextSize(48);
        textView.setTextColor(Color.BLACK);
        return textView;
    }
}

File: app/src/main/java/com/esafirm/androidplayground/conductor/ConductorSample.kt
package com.esafirm.androidplayground.conductor

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bluelinelabs.conductor.Controller
import com.bluelinelabs.conductor.changehandler.HorizontalChangeHandler
import com.esafirm.androidplayground.common.MenuFactory
import com.esafirm.androidplayground.conductor.sharedtransition.SharedTransitionController
import com.esafirm.conductorextra.transaction.Routes
import java.io.Serializable
import java.util.*

class ConductorSample : Controller(), Serializable {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup): View {
        return MenuFactory.create(container.context, Arrays.asList(
                "Configuration Change",
                "Shared Transition"
        )) { index ->
            when (index) {
                0 -> goToController(ConfigurationChangeController())
                1 -> goToController(SharedTransitionController())
            }
        }
    }

    private fun goToController(controller: Controller) {
        router.pushController(Routes.simpleTransaction(controller, HorizontalChangeHandler()))
    }
}
